### PR TITLE
Implement DELETE /users/{user_id endpoint for APIv4

### PR DESCRIPTION
**Pull request by ruzette**
#### Summary

Implement DELETE `/users/{user_id}` endpoint for APIv4.

#### Checklist

- [x] Added or updated unit tests (required for all new features)
- [x] Added API documentation (mattermost/mattermost-api-reference#81)
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

File: api4/user.go [hunk 1]
```diff
@@ -109,6 +110,33 @@ func updateUser(c *Context, w http.ResponseWriter, r *http.Request) {
...
+
+	if user, err = app.GetUser(c.Params.UserId); err != nil {
+		c.Err = err
+		return
+	}
+
+	if !app.SessionHasPermissionToUser(c.Session, user.Id) {
+		c.SetPermissionError(model.PERMISSION_EDIT_OTHER_USERS)
+		return
+	}
```
Comment (jwilander) [inline api4/user.go]: Minor, but can you move the permission check to be before getting the user?

Comment (ruzette) [inline api4/user.go]: Updated!

**Comment by jwilander** [comment]
Thanks for the changes @ruzette! Can you just give this a rebase against master to fix the conflicts and then I can get this merged?

**Comment by ruzette** [comment]
Hi @jwilander, I think I made too many conflict commits to fix to get rebase master to work. I can resubmit this PR along with the cherry-picked commits and resubmit, if that's more preferable?

**Comment by jwilander** [comment]
Hi @ruzette, yeah that's OK go for it

**Pull request closed without merging**